api/server.go

```diff
@@ -12,29 +12,55 @@ import (
 	"github.com/checkmarble/marble-backend/usecases"
 )
 
+type Option func(*options)
+
+func WithLocalTest(localTest bool) Option {
+	return func(o *options) {
+		o.localTest = localTest
+	}
+}
+
+type options struct {
+	localTest bool
+}
+
+func applyOptions(opts []Option) *options {
+	o := &options{
+		localTest: false,
+	}
+	for _, opt := range opts {
+		opt(o)
+	}
+	return o
+}
+
 func NewServer(
 	router *gin.Engine,
-	port string,
-	marbleAppHost string,
+	conf Configuration,
 	uc usecases.Usecases,
 	auth Authentication,
 	tokenHandler TokenHandler,
-	localTest ...bool,
+	opts ...Option,
 ) *http.Server {
-	addRoutes(router, auth, tokenHandler, uc, marbleAppHost)
+	o := applyOptions(opts)
+
+	addRoutes(router, conf, uc, auth, tokenHandler)
 
 	var host string
-	if len(localTest) > 0 && localTest[0] {
+	if o.localTest {
 		host = "localhost"
 	} else {
 		host = "0.0.0.0"
 	}
 
+	// Add 5 seconds to the server timeout to gracefully handle the timeout in our code
+	maxTimeout := max(conf.BatchTimeout, conf.DecisionTimeout, conf.DefaultTimeout) + 5*time.Second
```

Comment:
This is something new (based on our spec discussion) : IIUC the previous comment comming from 
```go
// Infra timeout is 60sec, so we set it to 55sec in order to gracefully handle the timeout in our code
	BATCH_INGESTION_TIMEOUT = 55 * time.Second
```
This makes sense to me

Reply:
Hum yea ok that works for me
The two kinds of timeout work on a different level but this should work fine